File: 01_Dev/web_projects/PY-JS-PHP/oop.js
// constructor and this keywords used

class Dog{
    constructor(name, breed){
        this.name = name;
        this.breed = breed;
    } //end constructor

    bark(){
        console.log("Woof! I'm " + this.name + "the "+ this.breed + ". ")
    } //end bark
} //end class

const myDog = new Dog("Bingo", "German Shepherd")
myDog.bark();

// Another class

class Car{
    constructor(color, model){
        this.color = color;
        this.model = model;
    } //end constructor

    start_car(){
        console.log("Starting...")
    }

    stop_car(){
        console.log("Stopping...")
    }
}

const myCar = new Car("red", "BMW");
myCar.start_car();
myCar.stop_car();

File: 01_Dev/web_projects/PY-JS-PHP/oop.py
# Simple class

class Dog:
    def __init__(self, name, breed):
        self.name = name
        self.breed = breed

    def bark(self):
        # simply print
        print(f"Woof! I'm {self.name} the {self.breed}" )

# Create an object from class
my_dog = Dog("Bingo", "German Shepherd")
my_dog.bark()

#Another class

class Car:
    def __init__(self, color, model):
        self.color = color
        self.model = model

    def start_car(self):
        print(f"Starting the {self.color} {self.model}")

    def stop_car(self):
        print(f"Stopping the {self.color} {self.model}")

my_car = Car("Black", "Jaguar")
my_car.start_car()
my_car.stop_car()
